### PR TITLE
Add WinGet Releaser

### PR DESCRIPTION
**Pull request by russellbanks**
This pull request adds a workflow that automatically publishes espanso to WinGet every release. *Please do not merge this pull request without reading the below and making the required changes* :)

- `token: ${{ secrets.WINGET_TOKEN }}`

This is the bit that will require action from the person whose account will be used for the pull requests (a member of the espanso GitHub organisation). This is a GitHub token with which the action will authenticate with GitHub and create a pull request on the [winget-pkgs](https://github.com/microsoft/winget-pkgs) repository.

To do this, you will need to:

1. Create a Personal Access Token (PAT) with `public_repo` scope.

image

2. Create a repository secret containing the token. **Super important: call this token: `WINGET_TOKEN`**.
See [using encrypted secrets in a workflow](https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow) for more details.

</br>

**Finally, a fork of [winget-pkgs](https://github.com/microsoft/winget-pkgs) needs to be created under the espanso organisation. Please do not delete the fork after a pull request has been merged as the action will use that fork for making a branch and merging it with the upstream [winget-pkgs](https://github.com/microsoft/winget-pkgs) repository on every espanso release.**

If you would like to read about this action further, the source code and documentation are [here](https://github.com/vedantmgoyal2009/winget-releaser).

Requesting review from @vedantmgoyal2009

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -107,6 +107,15 @@ jobs:
         run: |
           gh release upload ${{ needs.extract-version.outputs.espanso_version }} target/windows/installer/Espanso-Win-Installer-x86_64.exe target/windows/Espanso-Win-Portable-x86_64.zip target/windows/installer/Espanso-Win-Installer-x86_64.exe.sha256.txt target/windows/Espanso-Win-Portable-x86_64.zip.sha256.txt
   
+  windows-publish-winget:
+    needs: [ windows ]
```
Comment (vedantmgoyal2009) [inline .github/workflows/release.yml:111]:
```suggestion
  windows-publish-winget:
    if: ${{ github.ref == 'refs/heads/master' }}
    needs: [ windows ]
```

**Review by vedantmgoyal9 — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -107,6 +107,16 @@ jobs:
         run: |
           gh release upload ${{ needs.extract-version.outputs.espanso_version }} target/windows/installer/Espanso-Win-Installer-x86_64.exe target/windows/Espanso-Win-Portable-x86_64.zip target/windows/installer/Espanso-Win-Installer-x86_64.exe.sha256.txt target/windows/Espanso-Win-Portable-x86_64.zip.sha256.txt
   
+  windows-publish-winget:
+    if: ${{ github.ref == 'refs/heads/master' }}
+    needs: [ windows ]
+    runs-on: windows-latest
+    steps:
+      - uses: vedantmgoyal2009/winget-releaser@latest
```
Comment (vedantmgoyal2009) [inline .github/workflows/release.yml:115]:
```suggestion
      - uses: vedantmgoyal2009/winget-releaser@v1
```

**Review by vedantmgoyal9 — CHANGES_REQUESTED** [review]

**Comment by AucaCoyan** [comment]
Hi! interesting! there are a couple of conflicts to resolve. Can you make it? 💘  Thank you!

**Review by vedantmgoyal9 — CHANGES_REQUESTED** [review]
@russellbanks please resolve merge conflicts.

**Comment by federico-terzi** [comment]
Thanks! Before proceeding with this, I'd like to review how the Github Action for the release work :) I'll try to do it soon

Sorry for the late reply!

**Comment by AucaCoyan** [comment]
Closing by age. We can talk about it whenever you like!

**Pull request closed without merging**